### DISCUSSION TITLE
Plex LXC using another device as storage?

### DISCUSSION THREAD

**Discussion by Lockie85** [community-scripts/ProxmoxVE#336] [Q&A] [answered]
Hey,

I've got Plex LXC setup but I'd like to use the shared folder on my Synology NAS to store all forms of media. How can I add that devices folder to my plex LXC?

Many Thanks

**Comment by MickLesk** [comment]
Maybe this also works for Plex?
https://github.com/community-scripts/ProxmoxVE/discussions/175

**Reply by Lockie85** [reply, marked as answer]
This worked for me...

**Setting Up Plex LXC with Synology NAS Media via NFS on Proxmox**

This guide explains how to configure a Plex Media Server running in an unprivileged LXC container on Proxmox to access media stored on a Synology NAS via NFS. Replace placeholders like `<IP_ADDRESS>`, `<LXC_ID>`, and `<MOUNT_POINT>` with your specific details.

1. Prerequisites
- A Proxmox VE server with an unprivileged LXC running Plex.
- Synology NAS with shared media folder(s).
- Network connectivity between Proxmox, the Plex LXC, and the Synology NAS.

2. Configure NFS on Synology NAS

A. Enable NFS Service:
   - Log into your Synology NAS web interface.
   - Navigate to Control Panel > File Services > NFS Service.
   - Enable the NFS service and click Apply.

B. Set NFS Permissions for the Shared Folder:
   - Go to Control Panel > Shared Folder.
   - Select the shared folder (e.g., "media") and click Edit.
   - Go to the Permissions tab and open NFS Permissions.
   - Click Create and set the following:
     - Hostname or IP: <LXC_NETWORK_SUBNET> (e.g., `192.168.1.0/24`) or just the LXC IP.
     - Privilege: Read/Write.
     - Squash: Map all users to admin
     - Security: AUTH_SYS
     - Allow connections from non-privileged ports: Enable.
     - Allow users to access mounted subfolders: Enable.
     - Enable asynchronous: Enable.
   - Click OK to save the changes.

3. Configure the Proxmox Host

A. Install NFS Utilities on Proxmox:
SSH into your Proxmox server and ensure NFS tools are installed:
```
apt update
apt install nfs-common
```

B. Verify NFS Export:
Check if the NAS NFS share is accessible:
`showmount -e <NAS_IP>`

Example output:
```
Export list for <NAS_IP>:
/volume1/media <LXC_NETWORK_SUBNET>
```

C. Mount the NFS Share on Proxmox:
Create a mount point on the Proxmox host:
`mkdir -p /mnt/<MOUNT_POINT>`

Mount the NFS share:
   ```
   mount -t nfs <NAS_IP>:/volume1/media /mnt/<MOUNT_POINT>
   ```
   To make this permanent, add it to `/etc/fstab`:
   ```
   echo "<NAS_IP>:/volume1/media /mnt/<MOUNT_POINT> nfs defaults 0 0" >> /etc/fstab
   ```
   Test the configuration:
   ```
   mount -a
   ```

4. Bind-Mount NFS Share to the Plex LXC

A. Edit LXC Configuration:
   Open the LXC configuration file on the Proxmox host:
   ```
   nano /etc/pve/lxc/<LXC_ID>.conf
   ```
   Add a bind-mount entry for the NFS share:
   ```
   mp0: /mnt/<MOUNT_POINT>,mp=/mnt/<MOUNT_POINT>
   ```

B. Restart the Plex LXC:
   Restart the container to apply changes:
   ```
   pct stop <LXC_ID>
   pct start <LXC_ID>
   ```

C. Verify Mount in the LXC:
   Log into the LXC and check the mount point:
   ```
   pct exec <LXC_ID> -- bash
   ls /mnt/<MOUNT_POINT>
   ```
   You should see your media directories (e.g., `film`, `shows`).

5. Configure Plex Media Server

A. Open the Plex Web Interface:
   Navigate to the Plex web interface using your LXC's IP and port (default `32400`):
   ```
   http://<LXC_IP>:32400
   ```

B. Add Media Libraries:
   - Go to Settings > Manage > Libraries.
   - Click Add Library and select the type of media (e.g., Movies, TV Shows).
   - Click Browse for Media Folder and navigate to:
     ```
     /mnt/<MOUNT_POINT>/film
     ```
     or
     ```
     /mnt/<MOUNT_POINT>/shows
     ```
   - Save the library and let Plex scan the directory for media files.

**Reply by parithon** [reply]
The problem with this solution is that a local mount-point prevents the LXC from migrating to another host. :*(

**Reply by maxduracell** [reply]
When I do this everything works until I try to restart the lxc.  For some reason adding the mount point to the conf file breaks the container.  Did something with Proxmox change or am I simply doing something wrong?
